src/Core/Util/ConfigurationVariablesLoader.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace PrestaShop\PrestaShop\Core\Util;
+
+use Symfony\Component\Dotenv\Dotenv;
+
+class ConfigurationVariablesLoader
+{
+    private $loaded = false;
+
+    private const PARAMETERS_WITH_ENV_VARS = [
+        'database_host' => 'PS_DATABASE_HOST',
+        'database_port' => 'PS_DATABASE_PORT',
+        'database_user' => 'PS_DATABASE_USER',
+        'database_password' => 'PS_DATABASE_PASSWORD',
+        'database_name' => 'PS_DATABASE_NAME',
+        'database_prefix' => 'PS_DATABASE_PREFIX',
+        'database_engine' => 'PS_DATABASE_ENGINE',
+        'cookie_key' => 'PS_COOKIE_KEY',
+        'cookie_iv' => 'PS_COOKIE_IV',
+    ];
+
+    public function loadEnvVariables(array $parameters): array
+    {
+        if (!$this->loaded) {
+            (new Dotenv(true))->loadEnv(_PS_ROOT_DIR_ . '/.env');
+        }
+
+        if (!array_key_exists('parameters', $parameters)) {
+            return $parameters;
+        }
+
+        foreach ($parameters['parameters'] as $parameterKey => $parameterValue) {
+            if (isset(self::PARAMETERS_WITH_ENV_VARS[$parameterKey]) && false !== $envValue = getenv(self::PARAMETERS_WITH_ENV_VARS[$parameterKey])) {
+                $parameters['parameters'][$parameterKey] = $envValue;
+            }
```

Comment:
```suggestion
            $envKey = self::PARAMETERS_WITH_ENV_VARS[$parameterKey];
            if (isset($envKey) 
                && array_key_exists($envKey, $_ENV)
            ) {
                $parameters['parameters'][$parameterKey] = $_ENV[$envKey];
            }
```

Since we are using custom data, we can avoid `getenv` (which I recommended, but without we can have a false value :p)

It's also more readable.

Reply:
Agreed 😉  please dont mix assignment statements with conditions. It's very hard to read and maintain and prone to errors.